ACD-618 (#30)
Actindo-AG/Documentation: Channels/Integration/01e_Magento2.md

# Manage the Magento 2 connection

> To establish a connection to a Magento 2 shop, there are several particularities to consider. Therefore, the configuration of the Magento 2 connection is described in detail below.

Author: ```suggestion
To create a connection to a Magento 2 shop, there are no specific requirements. Only the Magento 2 driver must be installed in the main account of Actindo. For detailed information about creating a connection, see [Create a connection](./01_ManageConnections.md#create-a-connection).    
To configure the connection to a Magento 2 shop, there are several particularities to consider. Therefore, the configuration of the Magento 2 connection is described in detail below.
```

> To establish a connection to a Magento 2 shop, there are several particularities to consider. Therefore, the configuration of the Magento 2 connection is described in detail below.

Author: Warum haben wir hier nicht das Erstellen der Verbindung beschrieben? Brauche nochmal ne Erinnerung...

Reviewer: URL, Username und Password nötig. Nicht weiter erklärt in Wissenstransfer.

## Configure the Magento 2 connection   

After the connection to a Magento 2 shop has been established, further settings can be configured for the connection.

#### Prerequisites

> - A Magento 2 account has been created.
> - The *Magento 2* plugin has been installed.
> - A Magento 2 connection has been established.

Author: ```suggestion
A Magento 2 connection has been established.
```
WEnn die Verbindung hergestellt ist, müssen die anderen Voraussetzungen zwangsläufig erfüllt sein...

> [Info] For the *Omni-Channel* module version 4.1.0 or higher, the *Magento 2* plugin is required in at least version 1.0.0.

[comment]: <> (Stimmt das so? In Platform Manager Magento 2 1.1.1 Version und Magento 2 B.B. 1.0.0 verfügbar.)

#### Procedure

*Omni-Channel > Settings > Tab CONNECTIONS*

![Magento 2 connection](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/Connection.png "[Magento 2 connection]")

1. Click the Magento 2 connection in the list of connections.   

>   The *Edit connection* view is displayed. By default, the *Credentials* tab is displayed.

Author: ```suggestion
  The *Edit connection* view is displayed. By default, the *Credentials* tab is selected.
```

  ![Edit connection credentials](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/EditConnectionCredentials.png "[Edit connection credentials]")

2. Click the *Settings* tab.   

>   The *Settings* tab is displayed. By default, the *Store view languages* setting is displayed. All store views configured in the webshop are displayed in the right side bar.

Author: ```suggestion
  The *Settings* tab is displayed. By default, the *Store view languages* setting is selected. 
```

  ![Edit connection settings](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/EditConnectionSettings.png "[Edit connection settings]")

> 3. Click the desired *Store view* drop-down list and select the appropriate language. The available languages are displayed in the drop-down list.  

Author: ```suggestion
3. Click the *Store view* drop-down list and select the language in which the products will be uploaded to the store. All available languages are displayed in the list.  
```
Sofern du genauer spezifizieren kannst, was hier ausgewählt werden soll, solltest du das auch direkt hier beschreiben und nicht erst in einer Info, die eventuell niemand liest.
vgl. auch default sentences.

>     > [Info] The selected language determines the language of the products to be uploaded in the corresponding store view. If further languages are needed, they must be added in the *DataHub* module first, see [Manage the languages](../../../DataHub/Integration/05_ManageLanguages.md).

Author: ```suggestion
```

>     > [Info] The selected language determines the language of the products to be uploaded in the corresponding store view. If further languages are needed, they must be added in the *DataHub* module first, see [Manage the languages](../../../DataHub/Integration/05_ManageLanguages.md).
> 
> [comment]: <> (Stimmt das so?)

Author: ```suggestion
```

4. Click the *Order states* menu entry in the left side bar.  
    The order states and statuses are displayed in the right side bar.  

>     > [Info] An order state is a position of an order in the current processing flow. Order states are used interally and are not customizable. Each order state can have one or more order statuses assigned to it. Order statuses are customizable by the store administrator in the Magento system. 

Author: Woher kommt die Info? Ich verstehe das nicht so richtig...

Reviewer: Research aus dem Internet...

5. Configure the following settings to filter the orders to be imported in the *Filter orders to import* box:   

    - Enable the applicable toggle(s) to determine the order states. The following states are available:
        - *new*
        - *pending*
        - *pending_payment*
        - *payment_review*
        - *processing*
        - *holded*
        - *completed*
        - *closed*
        - *canceled*  

    - If desired, enter the applicable order number in the *Import only orders with a higher number that this* field.  

    - If desired, enter the applicable status in the *Status to filter for (leave empty if not required)*. If it is not required, this field can be left empty. By default, there are 13 statuses available:
        - *Suspect Fraud*
        - *Processing*
        - *Pending Payment*
        - *Payment Review*
        - *Pending*
        - *On Hold*
        - *Complete*
        - *Closed*
        - *Canceled*
        - *PayPal Canceled Reversal*
        - *Pending PayPal*
        - *PayPal Reversed*

>         - *Refunded*

Reviewer: Internet Research. S. Links oben

> 6. Configure the following settings to define the desired order statuses or states to be set after a certain event has occurred in the *Update order state and status on certain events* box:   

Author: ```suggestion
6. Configure the following settings to define the order states and/or statuses to be set after a certain event has occurred in the *Update order state and status on certain events* section:   
```

    
    - Enter the desired order status to be set when an order has been partially shipped in the *Order status to be set in shop after partially shipped* field.

    - Click the *Order state to be set in shop after fully shipped* drop-down list and select the desired state. The following states are available:  
        - **new**
        - **pending**
        - **pending_payment**
        - **payment_review**
        - **processing**
        - **holded**
        - **completed**
        - **closed**
        - **canceled**  

    - Enter the desired order status to be set when an order has been fully shipped in the *Order status to be set in shop after fully shipped* field.

    - Enter the desired order status to be set when an order has been partially paid in the *Order status to be set in shop after partially paid* field.

    - Enter the desired order status to be set when an order has been fully paid in the *Order status to be set in shop after fully paid* field.

    - Enter the desired order status to be set when a tracking number has been set for the order in the *Order status to be set in shop after tracking number set* field.

> 7.  If necessary, configure the following settings in the *Magento invoice handling* box:
> 
>     - Enable the *Create an invoice in Magento for orders that are fully paid* toggle. 
>     
>         > [Info] If enabled, invoice creation is triggered in Magento 2 as soon as an order is fully paid. This function is particularly relevant for the correct processing of vouchers.  
>     
>         The *Notify customer about the invoice* toggle is unlocked.
> 
>     - Enable the *Notify customer about the invoice* toggle to notify customers when an invoiced is generated.

Author: ```suggestion
7.  If necessary, configure the following settings in the *Magento invoice handling* section:

    + Enable the *Create an invoice in Magento for orders that are fully paid* toggle to create invoices in Magento 2 for fully paid orders and ensure the correct processing of vouchers.  
        
    + Enable the *Notify customer about the invoice* toggle to notify customers when an invoiced is generated. This toggle is only unlocked if the *Create an invoice in Magento for orders that are fully paid* toggle is enabled.
```
Ich muss nochmal checken, ob ich bislang in Unterpunkten auch results eingeschlossen habe. Ich meine, das habe ich bislang immer nur am Ende eines steps gemacht.

    

8. Click the *Tax classes* menu entry in the left side bar.  

>     The tax classes available in the *Actindo Core1 Platform* are displayed in the right side bar.

Author: ```suggestion
    All tax classes available in the *Actindo Core1 Platform* are displayed on the right side.
```

> 9. Click the *Select Magento 2 tax class for Core1 tax class* drop-down list and select the applicable Magento tax class. All available tax classes are displayed in the list. Repeat this step for all the applicable tax classes.

Author: ```suggestion
9. Click the *Select Magento 2 tax class for Core1 tax class* drop-down list and select the applicable Magento tax class. All tax classes available  in Magento 2 are displayed in the list. Repeat this step for all the applicable tax classes.
```

[comment]: <> (Gibt es Default tax classes bei Magento, z.B. die auf dem Video angezeigt oder abhängig von weiteren Vor-/Einstellungen in Magento? Neue tax classes können bei Magento auch erstellt werden, also: All available tax classes are displayed in the list oder By default, the following options are available?)

10. Click the *Global default language for Magento 2* menu entry in the left side bar.  

>     The *Select global default language for Magento backend* section is displayed in the right side bar.

Author: Einheitlichkeit! Mal sagst du, welche Felder alle angezeigt werden, mal welche section. Mir ist es eigentlich ziemlich egal, wir sollten uns aber auf eine Variante einigen!

11. Click the *Select Core1 language to be set as default* drop-down list and select the appropriate language. All languages available in the *Actindo Core1 Platform* are displayed.

> [comment]: <> (Unklar - RS mit Oli)

Author: nach Rücksprache mit Oli spezifizieren

12. Click the *Image handling* menu entry in the left side bar.  
    All available image attributes available in the Magento 2 gallery are displayed in the right side bar.

> 13. Enable the applicable toggles as necessary to hide or display the corresponding images in the Magento 2 gallery. 

Author: ```suggestion
13. Enable all toggles of the image attributes whose images you want to be hidden in the Magento 2 gallery. 
```

14. Click the *Image name* menu entry in the left side bar.  

>     The *Imagen name* settings are displayed are displayed in the right side bar.

Author: ```suggestion
    The *Image name* settings are displayed on the right side. By default, the *Apply from default* toggle is enabled and all fields are locked.
```

> 15. Disable the *Apply from default* toggle.  

Author: ```suggestion
15. Disable the *Apply from default* toggle to configure how the image names should be customized before uploading to the Magento 2 shop. 
```

>     The *Use ECM image name* toggle und the *Regular expressions* field are unlocked.

Author: ```suggestion
    The *Use ECM image name* toggle and the *Regular expressions* field are unlocked.
```

> 16. If necessary, disable the *Use ECM image name*.
> 
> 17. If desired, create regular expressions to define words or descriptions to be removed from the image name.  
> The site https://regex101.com/ can be useful to test regular expressions. 

Author: ```suggestion
16. If desired, disable the *Use ECM image name* toggle. By default, the toggle is enabled.

17. If desired, define regular expressions that will be removed from the image name.  
For detailed information about regular expressions, see https://regex101.com/.
```
Ist das regex Feld unabhängig vom toggle auswählbar? Welcher image name wird genommen wenn der toggle disabled ist?

>     > [Info] This function allows to create customized image names to make them SEO-friendly and improve search rankings.

Author: ```suggestion
    > [Info] The following settings are used to improve the SEO of your images.
```
Info nach step 15

    ![Settings image handling](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/EditConnectionSettings_ImageHandling.png "[Settings image handling]")

13. Enable all toggles of the image attributes whose images you want to be hidden in the Magento 2 gallery. 

14. Click the *Image name* menu entry in the left side bar.  
    The *Image name* settings are displayed are displayed on the right side.

    ![Settings image name](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/EditConnectionSettings_ImageName.png "[Settings image name]")

15. Disable the *Apply from default* toggle to configure how the image names should be customized before uploading to the Magento 2 shop and to improve the SEO of your images. Otherwise, the default setting is applied, that is, a randomly generated image name will be uploaded to the Magento 2 shop.   
    The *Use ECM image name* toggle and the *Regular expression* field are unlocked.

    ![Settings image name regular expression](../../Assets/Screenshots/Channels/Settings/Connections/Magento2/EditConnectionSettings_ImageNameRegEx.png "[Settings image name regular expression]")

> [comment]: <> (Oli: Random Name oder wird der ECM Name herangezogen?)

Reviewer 2: der ECM-Name wird verwendet falls die option drunter aus ist. Andernfalls wird ein zufälliger Dateiname verwendet
